server/src/handlers/pages.rs

```diff
@@ -19,13 +23,22 @@ pub async fn create_page(
         })
 }
 
+pub async fn get_pages(user: User, State(db): State<DB>) -> (StatusCode, Json<Vec<Page>>) {
+    match db.get_pages(&user.id) {
+        Ok(pages) => (StatusCode::OK, Json(pages)),
+        Err(_) => (StatusCode::INTERNAL_SERVER_ERROR, Json(Vec::new())),
+    }
+}
```

Comment:
Change the return type to `Result<Json<Vec<Page>>, StatusCode>`. The status code OK is implicit, and we don't need to return an empty vector when there is an error.